# **Version 1.1 V2:** Extracting Insights | Identifying Trends and Patterns Amongst Amazon Product Categories via SQL Queries
![amazon](https://media.giphy.com/media/qy3nxN4UJMb4c/giphy.gif)
## **By:** Alexandria Ransom
## **Project Preface | Why | Purpose | Value :** I will use basic SQL queries to uncover patterns and extract insights amongst 22 product categories. For an example, I will use basic SQL queries to explore different trend in product prices, identify which categories has the highest and lowest price products, and see the average price of products per categories. The purpose is to get a better understanding of overall product behavioramongst product categories. This project can be of value to Amazon FBA sellers to help support their decision making in decising which product they can add to their online store as an asset.  
### **Database Language:** SQLite3, Basic SQL Queries
### **Data Questions:** 


# **Import Required Modules**

In [1]:
import pandas as pd
import sqlite3

In [2]:
df = pd.read_csv("C:\\Users\\Alexandria\\Documents\\Ecom_Project_Data\\ecom_search_keywords.csv")

In [3]:
conn = sqlite3.connect("amazon.db")

In [4]:
#df.to_sql('keyterms', conn)

In [5]:
#pip install ipython-sql

In [6]:
%load_ext sql

In [7]:
%sql sqlite:///amazon.db

In [8]:
%%sql 

SELECT * 
FROM keyterms
LIMIT 10;

 * sqlite:///amazon.db
Done.


[(0, 'Organic Bath Bombs - 25 Extra Large Pcs + 3 Bubble Bars & Dried Flowers - Bubble Bath Shower Salts for Women Men & Kids - Relaxing Bathbombs Gift Set Bath Essentials Shower Steamers', 'Lagunamoon', 23.99, 3768.0, 125.0, 90394.0, '3', 14.6, 4550, 4.4, 9.39, 6475.0, 'Beauty & Personal Care', 'B085NJ7ZNN', 'https://www.amazon.com/dp/B085NJ7ZNN'),
 (1, 'Bath Bombs 7 Natural Bath Bomb Gift Set Handmade Bubble Bathbombs for Women Kids Shea Butter Moisturize Gifts for Mom Her Girlfriend Mothers Day Gifts for Birthday Valentines Christmas', 'STNTUS INNOVATIONS', 14.99, 3082.0, 102.0, 46199.0, '6', 7.6, 1546, 4.6, 7.39, 5648.0, 'Beauty & Personal Care', 'B07Q6BLNT6', 'https://www.amazon.com/dp/B07Q6BLNT6'),
 (2, 'LifeAround2Angels Bath Bombs Gift Set 12 USA made Fizzies Shea & Coco Butter Dry Skin Moisturize Perfect for Bubble & Spa Bath. Handmade Birthday Mothers day Gifts idea For Her/Him wife girlfriend', 'Lifearound2angels', 21.44, 11256.0, 375.0, 241329.0, '1', 12.43, 40630, 4.8, 9.01, 672.0, 'Beauty & Personal Care', 'B01MFGN8S5', 'https://www.amazon.com/dp/B01MFGN8S5'),
 (3, 'Nagaliving Bath Bombs Gift Set 10 Organic Bubble Bath Bombs Wonderful Fizz Effect Bath Gift for Valentines Day Christmas', 'nagaliving', 9.99, 4460.0, 148.0, 44555.0, '11', 4.05, 2634, 4.4, 5.94, 5041.0, 'Beauty & Personal Care', 'B081DGZP4B', 'https://www.amazon.com/dp/B081DGZP4B'),
 (4, 'Bath Bombs for Women - 24 All Natural BathBombs with Organic Essential Oils Moisturizing Shea Butter and Bath Salts for Relaxation and Stress Relief- Bath Bombs Gift Set - Gift for Kids Women', 'Mineral Me California', 25.45, 1638.0, 54.0, 41687.0, '8', 14.9, 2317, 4.7, 10.55, 9033.0, 'Beauty & Personal Care', 'B08G4DB4DB', 'https://www.amazon.com/dp/B08G4DB4DB'),
 (5, 'JOYIUS Bath Bombs Gift Set with Natural Essential Oils Fizzy Spa Moisturizes Dry Skin Bubble Baths Valentines Day Gift Ideas for Girlfriends Women Moms Wife Her/Him 6 x 4.0 Oz', 'JOYIUS', 7.99, 76.0, 2.0, 607.0, '4', 2.21, 254, 4.4, 5.78, 112946.0, 'Beauty & Personal Care', 'B087T2LKBT', 'https://www.amazon.com/dp/B087T2LKBT'),
 (6, "Organic Bath Bombs Gift Set 50 Handmade Bulk Bath Bombs For Kids Women Men Wonderful Fizz Effect Bath Gift For Valentine's Day Christmas & Any Anniversaries", 'nagaliving', 35.19, 1631.0, 54.0, 57395.0, '10', 19.45, 2708, 4.4, 15.74, 10140.0, 'Beauty & Personal Care', 'B07R8YFY4F', 'https://www.amazon.com/dp/B07R8YFY4F'),
 (7, '10 Bath Bombs for Kids Bath Bomb Gift Set Natural Bubble Bathbombs Shea Butter Dry Skin Moisturize Fizzy Spa Bath for Her Mom Women Kids Girls Girlfriend Birthday Christmas Valentines Mothers Day', 'comfso', 16.99, 1407.0, 46.0, 23905.0, '2', 9.3, 1104, 4.7, 7.69, 14111.0, 'Beauty & Personal Care', 'B08PFJ3T1Z', 'https://www.amazon.com/dp/B08PFJ3T1Z'),
 (8, "30 Pcs Bath Bomb Gift Set Natural Organic Mini Bath Bombs Handmade with Rich Fizz - Best Birthday Gift for Kids/Women/Men Mother's Day", 'PickMora', 18.39, 2324.0, 77.0, 42738.0, '2', 9.84, 3311, 4.5, 8.55, 5538.0, 'Beauty & Personal Care', 'B0883D2SC5', 'https://www.amazon.com/dp/B0883D2SC5'),
 (9, 'Aofmee Bath Bombs Gift Set Handmade Bubble and Floating Fizzies Spa Kit Shea and Cocoa Dry Skin Moisturize Birthday Valentines Mothers Day Anniversary Christmas Gifts for Women Mom Her Kids', 'Aofmee', 14.99, 2446.0, 81.0, 36666.0, '6', 7.6, 9952, 4.7, 7.39, 6169.0, 'Beauty & Personal Care', 'B07Q8XDQYJ', 'https://www.amazon.com/dp/B07Q8XDQYJ')]

In [9]:
%%sql 

ALTER TABLE keyterms
DROP COLUMN "Unnamed: 0"; -- removing unwanted columns

 * sqlite:///amazon.db
(sqlite3.OperationalError) no such column: "Unnamed: 0"
[SQL: ALTER TABLE keyterms
DROP COLUMN "Unnamed: 0"; -- removing unwanted columns]
(Background on this error at: https://sqlalche.me/e/14/e3q8)


In [10]:
%%sql
SELECT *
FROM keyterms
LIMIT 15;

 * sqlite:///amazon.db
Done.


[(0, 'Organic Bath Bombs - 25 Extra Large Pcs + 3 Bubble Bars & Dried Flowers - Bubble Bath Shower Salts for Women Men & Kids - Relaxing Bathbombs Gift Set Bath Essentials Shower Steamers', 'Lagunamoon', 23.99, 3768.0, 125.0, 90394.0, '3', 14.6, 4550, 4.4, 9.39, 6475.0, 'Beauty & Personal Care', 'B085NJ7ZNN', 'https://www.amazon.com/dp/B085NJ7ZNN'),
 (1, 'Bath Bombs 7 Natural Bath Bomb Gift Set Handmade Bubble Bathbombs for Women Kids Shea Butter Moisturize Gifts for Mom Her Girlfriend Mothers Day Gifts for Birthday Valentines Christmas', 'STNTUS INNOVATIONS', 14.99, 3082.0, 102.0, 46199.0, '6', 7.6, 1546, 4.6, 7.39, 5648.0, 'Beauty & Personal Care', 'B07Q6BLNT6', 'https://www.amazon.com/dp/B07Q6BLNT6'),
 (2, 'LifeAround2Angels Bath Bombs Gift Set 12 USA made Fizzies Shea & Coco Butter Dry Skin Moisturize Perfect for Bubble & Spa Bath. Handmade Birthday Mothers day Gifts idea For Her/Him wife girlfriend', 'Lifearound2angels', 21.44, 11256.0, 375.0, 241329.0, '1', 12.43, 40630, 4.8, 9.01, 672.0, 'Beauty & Personal Care', 'B01MFGN8S5', 'https://www.amazon.com/dp/B01MFGN8S5'),
 (3, 'Nagaliving Bath Bombs Gift Set 10 Organic Bubble Bath Bombs Wonderful Fizz Effect Bath Gift for Valentines Day Christmas', 'nagaliving', 9.99, 4460.0, 148.0, 44555.0, '11', 4.05, 2634, 4.4, 5.94, 5041.0, 'Beauty & Personal Care', 'B081DGZP4B', 'https://www.amazon.com/dp/B081DGZP4B'),
 (4, 'Bath Bombs for Women - 24 All Natural BathBombs with Organic Essential Oils Moisturizing Shea Butter and Bath Salts for Relaxation and Stress Relief- Bath Bombs Gift Set - Gift for Kids Women', 'Mineral Me California', 25.45, 1638.0, 54.0, 41687.0, '8', 14.9, 2317, 4.7, 10.55, 9033.0, 'Beauty & Personal Care', 'B08G4DB4DB', 'https://www.amazon.com/dp/B08G4DB4DB'),
 (5, 'JOYIUS Bath Bombs Gift Set with Natural Essential Oils Fizzy Spa Moisturizes Dry Skin Bubble Baths Valentines Day Gift Ideas for Girlfriends Women Moms Wife Her/Him 6 x 4.0 Oz', 'JOYIUS', 7.99, 76.0, 2.0, 607.0, '4', 2.21, 254, 4.4, 5.78, 112946.0, 'Beauty & Personal Care', 'B087T2LKBT', 'https://www.amazon.com/dp/B087T2LKBT'),
 (6, "Organic Bath Bombs Gift Set 50 Handmade Bulk Bath Bombs For Kids Women Men Wonderful Fizz Effect Bath Gift For Valentine's Day Christmas & Any Anniversaries", 'nagaliving', 35.19, 1631.0, 54.0, 57395.0, '10', 19.45, 2708, 4.4, 15.74, 10140.0, 'Beauty & Personal Care', 'B07R8YFY4F', 'https://www.amazon.com/dp/B07R8YFY4F'),
 (7, '10 Bath Bombs for Kids Bath Bomb Gift Set Natural Bubble Bathbombs Shea Butter Dry Skin Moisturize Fizzy Spa Bath for Her Mom Women Kids Girls Girlfriend Birthday Christmas Valentines Mothers Day', 'comfso', 16.99, 1407.0, 46.0, 23905.0, '2', 9.3, 1104, 4.7, 7.69, 14111.0, 'Beauty & Personal Care', 'B08PFJ3T1Z', 'https://www.amazon.com/dp/B08PFJ3T1Z'),
 (8, "30 Pcs Bath Bomb Gift Set Natural Organic Mini Bath Bombs Handmade with Rich Fizz - Best Birthday Gift for Kids/Women/Men Mother's Day", 'PickMora', 18.39, 2324.0, 77.0, 42738.0, '2', 9.84, 3311, 4.5, 8.55, 5538.0, 'Beauty & Personal Care', 'B0883D2SC5', 'https://www.amazon.com/dp/B0883D2SC5'),
 (9, 'Aofmee Bath Bombs Gift Set Handmade Bubble and Floating Fizzies Spa Kit Shea and Cocoa Dry Skin Moisturize Birthday Valentines Mothers Day Anniversary Christmas Gifts for Women Mom Her Kids', 'Aofmee', 14.99, 2446.0, 81.0, 36666.0, '6', 7.6, 9952, 4.7, 7.39, 6169.0, 'Beauty & Personal Care', 'B07Q8XDQYJ', 'https://www.amazon.com/dp/B07Q8XDQYJ'),
 (10, 'Bath Bombs Scented Candles Set Handmade Essential Oil Relaxing Bathbombs Bubble Spa. Bath Bombs for Women Gifts for Women. Fizzy to Moisturize Dry Skin (4 Bath Bomb+2 Candles)', 'LA BELLEFE', 18.99, 1610.0, 53.0, 30574.0, '11', 11.0, 4231, 4.7, 7.99, 8530.0, 'Beauty & Personal Care', 'B07K9R9H3W', 'https://www.amazon.com/dp/B07K9R9H3W'),
 (11, 'Bath Bombs Gift Set - 6 Relaxing Scents USA Made Bath Bombs for Women & Men Natural Bath Bombs for Kids Bath Bombs Bath Bombs for Girls Bathbomb Bath Sets for Women Gift Idea Spa Gifts for Women', 'Beauty by 

In [11]:
%%sql

SELECT DISTINCT ("category")
FROM keyterms
GROUP BY "category"

 * sqlite:///amazon.db
Done.


[('Amazon Launchpad',),
 ('Arts Crafts & Sewing',),
 ('Automotive',),
 ('Beauty & Personal Care',),
 ('Clothing Shoes & Jewelry',),
 ('Grocery & Gourmet Food',),
 ('Handmade Products',),
 ('Health & Household',),
 ('Home & Kitchen',),
 ('Industrial & Scientific',),
 ('Kindle Store',),
 ('Movies & TV',),
 ('Musical Instruments',),
 ('Office Products',),
 ('Patio Lawn & Garden',),
 ('Photo Background Shooting Tents',),
 ('Sports & Outdoors',),
 ('Tablet Cases',),
 ('Tablet Sleeves',),
 ('Tools & Home Improvement',),
 ('Toys & Games',),
 ('unknown',)]

**These are the total of 22 categories**

In [12]:
%%sql 

SELECT COUNT (DISTINCT category) AS number_of_categories
FROM keyterms;

 * sqlite:///amazon.db
Done.


[(22,)]

In [13]:
%%sql
SELECT * 
FROM keyterms
LIMIT 1;

 * sqlite:///amazon.db
Done.


[(0, 'Organic Bath Bombs - 25 Extra Large Pcs + 3 Bubble Bars & Dried Flowers - Bubble Bath Shower Salts for Women Men & Kids - Relaxing Bathbombs Gift Set Bath Essentials Shower Steamers', 'Lagunamoon', 23.99, 3768.0, 125.0, 90394.0, '3', 14.6, 4550, 4.4, 9.39, 6475.0, 'Beauty & Personal Care', 'B085NJ7ZNN', 'https://www.amazon.com/dp/B085NJ7ZNN')]

In [14]:
%%sql

SELECT "category",
       COUNT(product_name) AS total_products_in_category
FROM keyterms 
GROUP BY "category";

 * sqlite:///amazon.db
Done.


[('Amazon Launchpad', 2),
 ('Arts Crafts & Sewing', 240),
 ('Automotive', 2),
 ('Beauty & Personal Care', 2521),
 ('Clothing Shoes & Jewelry', 161),
 ('Grocery & Gourmet Food', 1),
 ('Handmade Products', 69),
 ('Health & Household', 13),
 ('Home & Kitchen', 12),
 ('Industrial & Scientific', 23),
 ('Kindle Store', 3),
 ('Movies & TV', 3),
 ('Musical Instruments', 1),
 ('Office Products', 748),
 ('Patio Lawn & Garden', 3),
 ('Photo Background Shooting Tents', 1),
 ('Sports & Outdoors', 99),
 ('Tablet Cases', 1),
 ('Tablet Sleeves', 2),
 ('Tools & Home Improvement', 596),
 ('Toys & Games', 7),
 ('unknown', 607)]

* Here it showing how many products throughout the 18 keywords belong to each category. 
* Beauty & Personal Care imported the most products in comaries to the category Tablet Cases with just 1 product. In terms of comparison between these two columns, we can only look at price but not compare them because one has more products than the other.
* Office products imported the second most products 748. 
* 

In [15]:
%%sql 

SELECT category,
      AVG("price") AS avg_price,
      AVG("net") AS avg_net, 
      SUM("daily_sales") AS daily_sales,
      AVG("rating") AS avg_rating
FROM keyterms
WHERE category = 'Beauty & Personal Care'
GROUP BY category; 


 * sqlite:///amazon.db
Done.


[('Beauty & Personal Care', 12.743343911146734, 6.970285600951964, 107322.0, 4.375525585085304)]

In [16]:
%%sql 

SELECT category, 
       COUNT(product_name) AS total_number_of_products,
       AVG(price) AS avg_price, 
       SUM(monthly_sales) AS avg_monthly_sales,
       SUM(daily_sales) AS total_daily_sales,
       SUM(monthly_revenue) AS total_monthly_revenue,
       AVG(net) AS avg_net,
       AVG(fees) AS avg_fees
FROM keyterms
GROUP BY category;

 * sqlite:///amazon.db
Done.


[('Amazon Launchpad', 2, 5.49, 296.0, 10.0, 3587.0, 1.51, 3.9799999999999995),
 ('Arts Crafts & Sewing', 240, 12.006458333333292, 316570.0, 16196.0, 3739920.5, 6.430416666666668, 5.535250000000002),
 ('Automotive', 2, 101.5, 1012.0, 33.0, 126247.0, 79.555, 21.945),
 ('Beauty & Personal Care', 2521, 12.743343911146734, 3240498.0, 107322.0, 32821577.5, 6.970285600951964, 5.3376913923046505),
 ('Clothing Shoes & Jewelry', 161, 20.716770186335356, 19137.0, 609.0, 301363.0, 10.69391304347825, 6.869627329192555),
 ('Grocery & Gourmet Food', 1, 10.99, 375.0, 12.0, 4121.0, 4.82, 6.17),
 ('Handmade Products', 69, 21.845652173913056, 296.0, 10.0, 3587.0, 8.117971014492749, 5.591014492753632),
 ('Health & Household', 13, 21.450000000000003, 202721.0, 6748.0, 5306233.0, 13.498461538461537, 8.94076923076923),
 ('Home & Kitchen', 12, 17.872500000000002, 138196.0, 5565.0, 1878683.0, 9.860833333333332, 6.619166666666666),
 ('Industrial & Scientific', 23, 24.51347826086956, 856.0, 23.0, 16146.0, 7.757391304347825, 7.7034782608695656),
 ('Kindle Store', 3, 7.04, 48.0, 1.0, 204.0, 6.260000000000001, 5.22),
 ('Movies & TV', 3, 28.74, 444.0, 15.0, 5380.5, 21.223333333333333, 7.516666666666667),
 ('Musical Instruments', 1, 7.99, 0.0, 0.0, 0.0, 3.81, 4.18),
 ('Office Products', 748, 14.980494652406318, 1169579.0, 42335.0, 11788105.5, 7.359705882352961, 6.094064171122984),
 ('Patio Lawn & Garden', 3, 537.9966666666667, 4765.0, 158.0, 177201.0, 266.1566666666667, 271.84),
 ('Photo Background Shooting Tents', 1, 105.99, 148.0, 5.0, 1793.5, 81.73, 24.26),
 ('Sports & Outdoors', 99, 21.651010101010097, 3303.0, 101.0, 59449.0, 12.490404040404039, 6.8865656565656606),
 ('Tablet Cases', 1, 10.99, 148.0, 5.0, 1793.5, 6.26, 5.22),
 ('Tablet Sleeves', 2, 10.99, 148.0, 10.0, 1793.5, 7.1, 3.89),
 ('Tools & Home Improvement', 596, 23.124110738254874, 1565253.0, 51912.0, 35136642.0, 14.537667785234925, 7.720117449664413),
 ('Toys & Games', 7, 39.120000000000005, 595.0, 53.0, 11820.0, 29.34857142857143, 10.698571428571428),
 ('unknown', 607, 15.247495881383738, 89392.0, 3020.0, 1083274.0, 6.857545304777633, 5.1651729818780705)]

In [ ]:
#%%sql figuring out the way to cast dates

#UPDATE keyterms

#SET date_first_available = CAST(date_first_available AS DATETIME)